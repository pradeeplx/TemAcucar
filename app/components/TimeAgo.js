import React from 'react'
import moment from 'moment'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Sentence from "./Sentence"

moment.locale('pt-br', {
  relativeTime : {
    future: "Em %s",
    past:   "Há %s",
    s:  "segundos",
    m:  "um minuto",
    mm: "%d minutos",
    h:  "uma hora",
    hh: "%d horas",
    d:  "um dia",
    dd: "%d dias",
    M:  "um mês",
    MM: "%d meses",
    y:  "um ano",
    yy: "%d anos",
  }
})

export default TimeAgo = ({ time, style, lowerCase }) => (
  <Sentence style={[{
    fontSize: 9 * fontFactor(),
  }, style]}>
    { lowerCase ? moment(time).fromNow().toLowerCase() : moment(time).fromNow() }
  </Sentence>
)
