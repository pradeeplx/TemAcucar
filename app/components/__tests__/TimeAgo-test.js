import 'react-native'
import React from 'react'
import TimeAgo from '../TimeAgo'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <TimeAgo
      style={{ top: 0 }}
      time={Date.now()}
      lowerCase={true}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
