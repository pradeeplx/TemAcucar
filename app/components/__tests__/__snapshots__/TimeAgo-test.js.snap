exports[`test renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#654C50",
        "fontFamily": "Avenir",
        "fontSize": 18,
      },
      Array [
        Object {
          "fontSize": 13.5,
        },
        Object {
          "top": 0,
        },
      ],
    ]
  }>
  hรก segundos
</Text>
`;
